### PR TITLE
.gitignore updates to ignore submodules

### PR DESCRIPTION
**Pull request by MichaelDimmitt**
This turns out to be an oddly hard problem.

I did not see any resources online with a solution that lets you ignore an entire folder based on the presence of a child file or child folder. In this case we would want to ignore any subfolders from tracking that contained .git folders.

So I went to the .gitignore docs. and learned you can do negation tracking with `!`
https://git-scm.com/docs/gitignore

There may be a better solution but this is the best I can come up with right now.
and should work for this project right now as long as we are mindful of whenever we need a new directory we need to opt into it in the gitignore by adding a negation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,20 +1,7 @@
...
-erl_crash.dump
-
-# Also ignore archive artifacts (built via "mix archive.build").
-*.ez
+**/*
+!*.*
+!bash
+!elixir
+!python
+!ruby
```
Comment (murjax) [inline .gitignore:6]: I wonder if we should eventually move these into a `current` folder so we don't have to do this for every iteration. I'm not planning on supporting more languages yet though so this is good enough.

**Review by murjax — APPROVED** [review]
Doesn't scale well but solves the problem for now. :+1:

**Comment by mwoods79** [comment]
If your subfolders have `.git` folders shouldn't you be using git submodules?

https://git-scm.com/docs/git-submodule

Using submodules would "negate" the need for this PR 🥇

**Comment by murjax** [comment]
The old Elixir project only has two commits. I'd also be fine simply importing the project and dropping the `.git` folder. Then we don't have to account for submodules at all.

**Comment by MichaelDimmitt** [comment]
1. this .gitignore thing goes away if we just have everything clone outside of the "GithubRepoCloner"
> example structure
> | -- Github-Repo-Cloner
> | -- murjax
> | -- michaeldimmitt

> previous: structure
> | -- Github-Repo-Cloner
> |-----> murjax
> |-----> michaeldimmitt

This makes sense to do anyway because if this repo was an app that someone would click to execute we would not be storing the folder inside that executable file. Instead, clicking the app would put the folder adjascent to the app you just clicked or to a designated path like $HOME/Desktop .

2. we may want to clone down all repos as we are doing now. 
but also make it easy to the fork repos that the user knows they want to change.

File: .gitignore (hunk 1 above)
Comment (MichaelDimmitt) [reply .gitignore:6]: `go` version is coming soon

**Comment by murjax** [comment]
I wonder if we should support an argument for destination path. Using relative paths to determine the clone location works if you're in one of the language projects. Some of these have executable binaries that could be added to someone's path. Providing a destination path seems less intrusive. 

Ultimately people should be checking git status before adding stuff to see what's staged.

Open to other ideas though.

**Comment by MichaelDimmitt** [comment]
> I wonder if we should support an argument for destination path. Using relative paths to determine the clone location works if you're in one of the language projects. Some of these have executable binaries that could be added to someone's path. Providing a destination path seems less intrusive.
> 
> Ultimately people should be checking git status before adding stuff to see what's staged.
> 
> Open to other ideas though.

Good point! I think this needs to be done as a flag
--path=/dalkndl/naslkdnlkn/alskdnlsn

and error handle if they put in a bad path.
1. do we fail the entire program and do no work prompting the user
2. do we tell the user but still download as a folder adjascent to this programs directory. @murjax

**Pull request merged**